vmm/src/device_manager.rs

```diff
@@ -4401,5 +4452,11 @@ impl Drop for DeviceManager {
         for (device, _, _) in self.virtio_devices.drain(..) {
             device.lock().unwrap().shutdown();
         }
+        if let Some(serial_kill_evt) = &self.serial_kill_evt {
+            info!(
+                "stopping serial-manager thread: {:?}",
+                serial_kill_evt.write(1)
+            );
+        }
```

Comment:
Now `.join` on the thread handle to wait for the thread to finish.

Reply:
Ah right, I completely forgot about this as I went through a few restructures X(.